be/src/storage/rowset/segment_iterator.cpp

```diff
@@ -179,6 +191,8 @@ class SegmentIterator final : public ChunkIterator {
     template <bool check_global_dict>
     Status _init_column_iterators(const Schema& schema);
     Status _get_row_ranges_by_keys();
+    Status _get_row_ranges_by_key_ranges();
+    Status _get_row_ranges_by_short_key_ranges();
```

Comment:
What's the main difference between `_get_row_ranges_by_key_ranges` and `_get_row_ranges_by_short_key_ranges `?

Reply:
`_get_row_ranges_by_key_ranges()` uses key ranges whose endpoint is the completed `SeekTuple`, while `_get_row_ranges_by_short_key_ranges()` uses key ranges whose endpoint may be  completed `SeekTuple` or short key `Slice` from the segment index.